aws/resource_aws_resource_group.go

```diff
@@ -0,0 +1,169 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/resourcegroups"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsResourceGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsResourceGroupCreate,
+		Read:   resourceAwsResourceGroupRead,
+		Update: resourceAwsResourceGroupUpdate,
+		Delete: resourceAwsResourceGroupDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"resource_query": {
+				Type:     schema.TypeSet,
```

Comment:
For "container" attributes with `MaxItems: 1` we prefer to use `TypeList` instead of `TypeSet` to simplify a lot of things. 👍 

```suggestion
				Type:     schema.TypeList,
```